src/outputs/parthenon_hdf5.hpp

```diff
@@ -102,6 +102,9 @@ static hid_t getHDF5Type(const int32_t *) { return H5T_NATIVE_INT32; }
 static hid_t getHDF5Type(const int64_t *) { return H5T_NATIVE_INT64; }
 static hid_t getHDF5Type(const uint32_t *) { return H5T_NATIVE_UINT32; }
 static hid_t getHDF5Type(const uint64_t *) { return H5T_NATIVE_UINT64; }
+#ifdef __APPLE__
+static hid_t getHDF5Type(const size_t *) { return H5T_NATIVE_UINT64; }
+#endif
```

Comment:
```suggestion
static hid_t getHDF5Type(const size_t *) { return H5T_NATIVE_UINT64; }
```

This is my suggestion. Basically, making sure that our `size_t` is the same on all systems from an hdf5 point of view (so no extra switch for OSX).

Reply:
Ah I see. For HDF5 I think this is reasonable... Ideally the types on disk *should* be the same for all platforms

Reply:
Actually, this suggestion won't work as it results in two identical definitions on non-Apple arch so we may need to keep this.
However, I'm currently leaning towards using a template approach (based on checking if `size_t` is `uint64_t`) so that this is compiler and architecture independent (though I cannot test due to lack of access to OSX).